specification/keyvault/data-plane/TypeSpec/routes.tsp

```diff
@@ -0,0 +1,278 @@
+    @body
+    @flattenProperty
+    parameters: SecretSetParameters;
+  },
+  SecretBundle
+>;
+
+/**
+ * The DELETE operation applies to any secret stored in Azure Key Vault. DELETE
+ * cannot be applied to an individual version of a secret. This operation requires
+ * the secrets/delete permission.
+ */
+#suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Foundations.Operation is necessary for Key Vault"
+@summary("Deletes a secret from a specified key vault.")
+@route("/secrets/{secretName}")
+@delete
+op deleteSecret is Azure.Core.Foundations.Operation<
+  {
+    /**
+     * The name of the secret.
+     */
+    @path
+    secretName: string;
+  },
+  DeletedSecretBundle
+>;
+
+/**
+ * The UPDATE operation changes specified attributes of an existing stored secret.
+ * Attributes that are not specified in the request are left unchanged. The value
+ * of a secret itself cannot be changed. This operation requires the secrets/set
+ * permission.
+ */
+#suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Foundations.Operation is necessary for Key Vault"
+@summary("Updates the attributes associated with a specified secret in a given key vault.")
+@route("/secrets/{secretName}/{secretVersion}")
+@patch
+op updateSecret is Azure.Core.Foundations.Operation<
+  {
+    /**
+     * The name of the secret.
+     */
+    @path
+    secretName: string;
+
+    /**
+     * The version of the secret.
+     */
+    @path
+    secretVersion: string;
+
+    /**
+     * The parameters for update secret operation.
+     */
+    #suppress "deprecated" "Property flattening is supported for legacy scenarios like Key Vault's"
+    @body
+    @flattenProperty
+    parameters: SecretUpdateParameters;
+  },
+  SecretBundle
+>;
+
+/**
+ * The GET operation is applicable to any secret stored in Azure Key Vault. This
+ * operation requires the secrets/get permission.
+ */
+#suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Foundations.Operation is necessary for Key Vault"
+@summary("Get a specified secret from a given key vault.")
+@route("/secrets/{secretName}/{secretVersion}")
+@get
+op getSecret is Azure.Core.Foundations.Operation<
+  {
+    /**
+     * The name of the secret.
+     */
+    @path
+    secretName: string;
+
+    /**
+     * The version of the secret. This URI fragment is optional. If not specified, the
+     * latest version of the secret is returned.
+     */
+    @path
+    secretVersion: string;
+  },
+  SecretBundle
+>;
+
+/**
+ * The Get Secrets operation is applicable to the entire vault. However, only the
+ * base secret identifier and its attributes are provided in the response.
+ * Individual secret versions are not listed in the response. This operation
+ * requires the secrets/list permission.
+ */
+@summary("List secrets in a specified key vault.")
+op getSecrets is KeyVaultOperations.ResourceList<
```

Comment:
This uses the right pattern. Why don't the others? Do they not work? Do we need to declare our own traits so they do align? Key Vault doesn't really support many of the (now-) standard params.

Reply:
Answered in https://github.com/Azure/azure-rest-api-specs/pull/29249#discussion_r1635638503